### DISCUSSION TITLE
Nodejs version mismatch during deployment

### DISCUSSION THREAD

**Discussion by Uri3lArchangel** [vercel/next.js#45995] [Help]
Hi, so i deployed a nextjs project from my github and i kept getting this error
"error manzo-next@1.0.0: The engine "node" is incompatible with this module. Expected version "18.0.0". Got "18.12.1""
now my package.json file has the following ;

{
  "name": "manzo-next",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start -p $PORT",
    "lint": "next lint"
  },
  "dependencies": {
    "@ant-design/icons": "^4.7.0",
    "antd": "^4.18.2",
    "antd-img-crop": "^4.1.0",
    "ethers": "^5.5.3",
    "moment": "^2.29.1",
    "next": "12.0.7",
    "nvm": "^0.0.4",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-google-recaptcha-v3": "^1.9.7",
    "sass": "^1.58.1"
  },
  "devDependencies": {
    "eslint": "8.6.0",
    "eslint-config-next": "12.0.7"
  },
  "engines": {
    "node": "18.0.0",
    "npm": "8.19.2"
  },
  "description": "<p align=\"center\">   <a href=\"https://directprivateoffers.ca/\" target=\"blank\"><img src=\"./public/assets/images/directprivateoffers-logo-bd.png\" width=\"320\" alt=\"J&D Landscaping Logo\" /></a> </p>",
  "version": "1.0.0",
  "main": "next.config.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Uri3lArchangel/DPO-dev-frontend.git"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Uri3lArchangel/DPO-dev-frontend/issues"
  },
  "homepage": "https://github.com/Uri3lArchangel/DPO-dev-frontend#readme"
}

Please i need to know where 8.12.1 is coming from because i have also configured my project settings to run  with node 18.X.X

**Comment by icyJoseph** [comment]
The 18.12.1 is most likely coming from the environment provider, like Vercel for example gives 16.x.x and 18.x.x IIRC, https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js:

I think, you need to loosen up the engines declaration in your package.json:

```json
{
  "engines": {
    "node": ">=18.0.0"
  }
}
```

So that anything that's above 18.0.0, inclusive, is allowed.
